astrokat/noisediode.py

```diff
@@ -68,13 +73,21 @@ def on(kat,
     user_logger.trace('TRACE: nd on at {} ({})'
                       .format(timestamp,
                               time.ctime(timestamp)))
-    msg = ('Switch noise-diode on at {}'
+    msg = ('Request: Switch noise-diode on at {}'
            .format(timestamp))
     user_logger.info(msg)
 
     replies = kat.ants.req.dig_noise_source(timestamp, 1)
     if not kat.dry_run:
-        _katcp_reply_to_log_(replies)
+        timestamp = _katcp_reply_to_log_(replies)
+    else:
+        # - use integer second boundary as that is most likely be an exact
+        #   time that DMC can execute at
```

Comment:
This comment is more appropriate around line 72 - that is where apply the `ceil` to the time.  And it should happen regardless of the dry-run state.